tests/auth/ldap_test.py

```diff
@@ -6,33 +6,92 @@
 """
 import pytest
 
+# module must have ldap.INVALID_CREDENTIALS
+INVALID_CREDENTIALS = Exception
+
 
 def test_happy_path():
-    from sirepo import srunit
     from pykern.pkcollections import PKDict
     from pykern import pkinspect
+    from sirepo import srunit
     import sys
 
     sys.modules["ldap"] = pkinspect.this_module()
```

Comment:
This code is copied three times. It's very specific and hard to get right as well as distracting from the core part of the test. Refactor.